lib/posix/spawn/child.rb

```diff
@@ -262,12 +294,10 @@ def read_and_write(input, stdin, stdout, stderr, timeout=nil, max=nil)
           end
 
           # maybe we've hit our max output
-          if max && ready[0].any? && (@out.size + @err.size) > max
+          if max && ready[0].any? && bytes_seen > max
             raise MaximumOutputExceeded
           end
         end
-
-        [@out, @err]
```

Comment:
I decided to just drop returning these in an attempt at consistency since these one or both of these ivars are useless if we're streaming.